### PR TITLE
Add io.github.dfandrich.gpscorrelate

### PR DESCRIPTION
**Pull request by dfandrich**

### CODE REVIEW

File: io.github.dfandrich.gpscorrelate.json [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            "name": "gpscorrelate-gui",
+            "buildsystem": "simple",
+            "build-commands": [
+				/* gettext catalogues are dropped if stored in the standard location, so move them */
+                "make -j${FLATPAK_BUILDER_N_JOBS} prefix=${FLATPAK_DEST} localedir=${FLATPAK_DEST}/share/gpscorrelate/locale CFLAGS=\"${CFLAGS} -DENABLE_NLS=1\"",
+                "make prefix=${FLATPAK_DEST} localedir=${FLATPAK_DEST}/share/gpscorrelate/locale install install-po install-desktop-file",
+                "make check",
+                "mv ${FLATPAK_DEST}/share/icons/hicolor/scalable/apps/gpscorrelate-gui.svg ${FLATPAK_DEST}/share/icons/hicolor/scalable/apps/io.github.dfandrich.gpscorrelate.svg",
+                "mv ${FLATPAK_DEST}/share/applications/gpscorrelate.desktop ${FLATPAK_DEST}/share/applications/io.github.dfandrich.gpscorrelate.desktop",
+                "sed -i 's/Icon=gpscorrelate-gui/Icon=io.github.dfandrich.gpscorrelate/' ${FLATPAK_DEST}/share/applications/io.github.dfandrich.gpscorrelate.desktop"
```
Comment (bilelmoussaoui) [inline io.github.dfandrich.gpscorrelate.json:49]: use desktop-file-edit instead

File: io.github.dfandrich.gpscorrelate.json [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	{
+            "name": "gpscorrelate-gui",
+            "buildsystem": "simple",
+            "build-commands": [
+				/* gettext catalogues are dropped if stored in the standard location, so move them */
+                "make -j${FLATPAK_BUILDER_N_JOBS} prefix=${FLATPAK_DEST} localedir=${FLATPAK_DEST}/share/gpscorrelate/locale CFLAGS=\"${CFLAGS} -DENABLE_NLS=1\"",
+                "make prefix=${FLATPAK_DEST} localedir=${FLATPAK_DEST}/share/gpscorrelate/locale install install-po install-desktop-file",
+                "make check",
+                "mv ${FLATPAK_DEST}/share/icons/hicolor/scalable/apps/gpscorrelate-gui.svg ${FLATPAK_DEST}/share/icons/hicolor/scalable/apps/io.github.dfandrich.gpscorrelate.svg",
+                "mv ${FLATPAK_DEST}/share/applications/gpscorrelate.desktop ${FLATPAK_DEST}/share/applications/io.github.dfandrich.gpscorrelate.desktop",
```
Comment (bilelmoussaoui) [inline io.github.dfandrich.gpscorrelate.json:48]: use install instead of mv

File: io.github.dfandrich.gpscorrelate.json [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+                "/share/man"
+            ],
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://www.exiv2.org/builds/exiv2-0.27.3-Source.tar.gz",
+                    "sha256": "a79f5613812aa21755d578a297874fb59a85101e793edc64ec2c6bd994e3e778"
+                }]
+    },
+	{
```
Comment (bilelmoussaoui) [inline io.github.dfandrich.gpscorrelate.json:39]: The whole modules seems to be mis-indented

File: io.github.dfandrich.gpscorrelate.metainfo.xml [hunk 4]
```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (bilelmoussaoui) [inline io.github.dfandrich.gpscorrelate.metainfo.xml:1]: Should be sent upstream if possible

File: io.github.dfandrich.gpscorrelate.json [hunk 5]
```diff
@@ -0,0 +1,63 @@
+{
+  "app-id":"io.github.dfandrich.gpscorrelate",
+  "runtime":"org.freedesktop.Platform",
+  "runtime-version":"20.08",
+  "sdk":"org.freedesktop.Sdk",
+  "command":"gpscorrelate-gui",
+  "finish-args":[
+    "--socket=fallback-x11",
```
Comment (bilelmoussaoui) [inline io.github.dfandrich.gpscorrelate.json:8]: You need `"--share=ipc"` as well for x11

File: io.github.dfandrich.gpscorrelate.json [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    "--socket=wayland",
+	"--filesystem=home",
```
Comment (bilelmoussaoui) [inline io.github.dfandrich.gpscorrelate.json:10]: Please re-indent this; also, what does it need home access for?

**Comment by bilelmoussaoui** [comment]
bot, build io.github.dfandrich.gpscorrelate

File: io.github.dfandrich.gpscorrelate.metainfo.xml (hunk 4 above)
Comment (dfandrich) [reply io.github.dfandrich.gpscorrelate.metainfo.xml:1]: It's in process, but I wanted to build the current version right now.

File: io.github.dfandrich.gpscorrelate.json (hunk 5 above)
Comment (dfandrich) [reply io.github.dfandrich.gpscorrelate.json:8]: Done.

File: io.github.dfandrich.gpscorrelate.json (hunk 2 above)
Comment (dfandrich) [reply io.github.dfandrich.gpscorrelate.json:48]: I've reworked the install process.

File: io.github.dfandrich.gpscorrelate.json (hunk 1 above)
Comment (dfandrich) [reply io.github.dfandrich.gpscorrelate.json:49]: Using desktop-file-install now.

File: io.github.dfandrich.gpscorrelate.json (hunk 3 above)
Comment (dfandrich) [reply io.github.dfandrich.gpscorrelate.json:39]: Fixed. There were a few tabs hidden in there, too.

File: io.github.dfandrich.gpscorrelate.json (hunk 6 above)
Comment (dfandrich) [reply io.github.dfandrich.gpscorrelate.json:10]: I wasn't seeing any files from the chooser without --filesystem=home. But I realize now that there must be something else going on there. The code is using gtk_file_chooser_dialog_new which I assume should be working.

Comment (bilelmoussaoui) [reply io.github.dfandrich.gpscorrelate.json:10]: You need to use a native_file_chooser to use the portals.

**Comment by dfandrich** [comment]
This script is now also copying the metainfo file into place which I hope will fix the test build.

**Comment by bilelmoussaoui** [comment]
bot, build io.github.dfandrich.gpscorrelate

File: io.github.dfandrich.gpscorrelate.json [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+            "name": "gpscorrelate-gui",
+            "buildsystem": "simple",
+            "build-commands": [
+                /* gettext catalogues are dropped if stored in the standard location, so move them */
+                "make -j${FLATPAK_BUILDER_N_JOBS} prefix=${FLATPAK_DEST} localedir=${FLATPAK_DEST}/share/gpscorrelate/locale CFLAGS=\"${CFLAGS} -DENABLE_NLS=1\"",
+                "make prefix=${FLATPAK_DEST} localedir=${FLATPAK_DEST}/share/gpscorrelate/locale install install-po",
+                "cp gpscorrelate.desktop ${FLATPAK_ID}.desktop",
+                "desktop-file-install --vendor= --set-icon=${FLATPAK_ID} --dir=/app/share/applications ${FLATPAK_ID}.desktop",
+                "install -m0644 -Dp gpscorrelate-gui.svg ${FLATPAK_DEST}/share/icons/hicolor/scalable/apps/${FLATPAK_ID}.svg",
+                "ls -l",
```
Comment (bilelmoussaoui) [inline io.github.dfandrich.gpscorrelate.json:49]:
```suggestion
```

Comment (dfandrich) [reply io.github.dfandrich.gpscorrelate.json]: Removed.

**Comment by dfandrich** [comment]
I've added an upstream exiv2 patch that should fix the ARM build. I've also added the filesystem capability for the moment until the application can use a file chooser portal.

**Comment by dfandrich** [comment]
bot, build io.github.dfandrich.gpscorrelate

**Comment by dfandrich** [comment]
Updated the patch for aarch64.

bot, build io.github.dfandrich.gpscorrelate

**Pull request closed without merging**